backend/Testing/FwHeadless/MediaFileTests.cs

```diff
@@ -0,0 +1,341 @@
+        metadata.ExtraFields["one"].Should().BeOfType<JsonElement>();
+        ((JsonElement)metadata.ExtraFields["one"]).GetString().Should().Be("two");
+        metadata.ExtraFields.Should().ContainKey("three");
+        metadata.ExtraFields["three"].Should().BeOfType<JsonElement>();
+        ((JsonElement)metadata.ExtraFields["three"]).GetString().Should().Be("four");
+    }
+
+    [Fact]
+    public async Task UploadFile_WithNoFilenameField_FilenameTakenFromUploadedFile()
+    {
+        var (fileId, result) = await Fixture.PostFile(TestRepoZipPath, loginAs: "admin");
+        result.StatusCode.Should().Be(HttpStatusCode.Created);
+        var (metadata, mResult) = await Fixture.GetFileMetadata(fileId, loginAs: "admin");
+        mResult.StatusCode.Should().Be(HttpStatusCode.OK);
+        metadata.Should().NotBeNull();
+        metadata.Filename.Should().Be(TestRepoZipFilename);
+    }
+
+    [Fact]
+    public async Task UploadFile_TwiceWithDifferentFilenames_FilenameTakenFromFirstUpload()
+    {
+        var (fileId, result) = await Fixture.PostFile(TestRepoZipPath);
+        result.StatusCode.Should().Be(HttpStatusCode.Created);
+        var secondPath = TestRepoZipPath + ".bak";
+        if (File.Exists(secondPath)) File.Delete(secondPath);
+        File.Copy(TestRepoZipPath, secondPath);
+        result = await Fixture.PutFile(secondPath, fileId);
+        result.StatusCode.Should().Be(HttpStatusCode.OK);
+        var (metadata, mResult) = await Fixture.GetFileMetadata(fileId, loginAs: "admin");
+        mResult.StatusCode.Should().Be(HttpStatusCode.OK);
+        metadata.Should().NotBeNull();
+        metadata.Filename.Should().Be(TestRepoZipFilename);
+        var (files, listResult) = await Fixture.ListFiles(Fixture.ProjectId, loginAs: "admin");
+        listResult.StatusCode.Should().Be(HttpStatusCode.OK);
+        (files?.Files ?? []).Should().Contain(Path.Join(fileId.ToString(), TestRepoZipFilename));
+        (files?.Files ?? []).Should().NotContain(Path.Join(fileId.ToString(), secondPath));
+    }
+
+    [Fact]
+    public async Task UploadFile_TwiceWithDifferentFilenamesButOverridingFilename_Works()
+    {
+        var (fileId, result) = await Fixture.PostFile(TestRepoZipPath);
+        result.StatusCode.Should().Be(HttpStatusCode.Created);
+        var secondPath = TestRepoZipPath + ".bak";
+        if (File.Exists(secondPath)) File.Delete(secondPath);
+        File.Copy(TestRepoZipPath, secondPath);
+        result = await Fixture.PutFile(secondPath, fileId, overrideFilename: TestRepoZipFilename);
+        result.StatusCode.Should().Be(HttpStatusCode.OK);
+        var (metadata, mResult) = await Fixture.GetFileMetadata(fileId, loginAs: "admin");
+        mResult.StatusCode.Should().Be(HttpStatusCode.OK);
+        metadata.Should().NotBeNull();
+        metadata.Filename.Should().Be(TestRepoZipFilename);
+    }
+
+    [Fact]
+    public async Task UploadFile_TooLarge_ThrowsError()
```

Comment:
in the backend there's a couple ways we determine the file length, we check the content length header, the form file length, and last is the final length on disk. We should have a test for each of those cases. I would assume this one is testing the content length or the form file length.